From: David Steele <dsteele@gmail.com>
Date: Wed, 20 Nov 2013 21:52:37 -0500
Subject: Refresh XMPP authentication in process_jobs()

... if the username and password is known, to avoid problems with
the SASL token timing out.
---
 cloudprint/cloudprint.py | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/cloudprint/cloudprint.py b/cloudprint/cloudprint.py
index daf2606..a5b622c 100755
--- a/cloudprint/cloudprint.py
+++ b/cloudprint/cloudprint.py
@@ -400,6 +400,15 @@ def process_jobs(cups_connection, cpp, printers):
             LOGGER.error('ERROR: Could not Connect to Cloud Service. Will Try again in %d Seconds' % FAIL_RETRY)
             time.sleep(FAIL_RETRY)
 
+        if cpp.username and not xmpp_conn.is_connected():
+            LOGGER.debug('Refreshing authentication')
+            cpp.set_auth('')
+            try:
+                cpp.get_auth()
+                xmpp_auth = file(cpp.xmpp_auth_path).read()
+            except:
+                LOGGER.debug('Error refreshing authentication')
+
 
 def usage():
     print sys.argv[0] + ' [-d][-l][-h][-c][-f][-u][-v] [-p pid_file] [-a account_file]'
